study1/week3/evaluate-the-bracket-pairs-of-a-string.java

```diff
@@ -0,0 +1,42 @@
+//https://leetcode.com/problems/evaluate-the-bracket-pairs-of-a-string/description/
+
+class Solution {
+    public String evaluate(String s, List<List<String>> knowledge) {
+        //time complexity is O(n), if the length of s is n
+        //space complexity is O(n), if the number of keys in knowledge is n
+
+        //set new data structure, to make easier to find values
+        Map<String, String> map = new HashMap<String, String>();
+
+        for (List<String> list : knowledge) {
+            map.put(list.get(0), list.get(1));
+        }
+
+        StringBuilder sb = new StringBuilder(); //for return string
```

Comment:
I like that you used `StringBuilder` here. Can you explain how the complexity would change when you use `String` instead here?